### PR TITLE
Multichain discovery and UpdateMonitor

### PR DESCRIPTION
**Pull request by antooni**
This PR will:
- enable discovery on multiple chains
- enable update-monitor to work on multiple chains
- introduce BREAKING CHANGES:
  - make sure to update you .env variables
  - discovery script will now run via `yarn discover <chain> <project>`
- update configHashes of all projects (because the chainId field was added) 

Resolves L2B-1967

The most important files to check are:
- ConfigReader - is the file parsing correct?
- UpdateMonitor - is this idea with updateChain function great?
- DiscoveryRunner - is passing provider and engine a good pattern?
- cli - try running the discovery ;p
- ChainId - is this .getName() okay?

### CODE REVIEW

**Review by mateuszradomski — CHANGES_REQUESTED** [review]
Nice job, a few little things to discuss only as I see it :))

Also one small annoyance when you run discovery without any arguments the usage that get's printed does not mention chainId

```
$ discovery discover
Error: Not enough arguments

Usage:
    yarn start ...................................... run the server application
    yarn discover [project] ................. run discovery on a specific system
    yarn discover [project] --dry-run ..... check simulated discovery bot output
    yarn discover [project] --dev ..... run on the same block number as in .json
    yarn invert [project] ..................... print addresses and their functions
    yarn invert [project] --mermaid .................... print mermaid graph markup
    yarn <start|discover> --help .......................... display this message
```

File: packages/backend/src/api/routers/StatusRouter.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export function createStatusRouter(statusController: StatusController) {
   )
 
   router.get('/status/discovery', async (ctx) => {
-    ctx.body = await statusController.getDiscoveryDashboard()
+    ctx.body = await statusController.getDiscoveryDashboard(ChainId.ETHEREUM)
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/backend/src/api/routers/StatusRouter.ts:81]: there is a task to update dashboard for multiple chains

File: packages/backend/discovery/base/config.jsonc
[the remark this answers is not in the document]
Comment (lucadonnoh) [reply packages/backend/discovery/base/config.jsonc:1]: maybe i pushed `base/config.jsonc` by mistake? idk

File: packages/shared-pure/src/types/ChainId.ts [hunk 2]
```diff
@@ -40,3 +40,15 @@ ChainId.getName = function (chainId: ChainId): string {
...
+
+ChainId.getId = function (chainName: string): ChainId {
+  const chain = CHAIN_IDS.find((c) => c.name === chainName)
+  if (!chain) {
+    throw new TypeError('Invalid ChainId')
+  }
+  return ChainId(chain.id)
+}
+
+ChainId.getAll = function (): ChainId[] {
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/shared-pure/src/types/ChainId.ts:52]: we use it in ConfigReader tests

File: packages/discovery/src/discovery/config/DiscoveryConfig.ts [hunk 3]
```diff
@@ -21,6 +26,10 @@ export class DiscoveryConfig {
     return this.config.name
   }
 
+  get chainId() {
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/discovery/src/discovery/config/DiscoveryConfig.ts:29]: yeah, good idea, then the misconfiguration will come up in the tests

**Comment by antooni** [comment]
> Nice job, a few little things to discuss only as I see it :))
> 
> Also one small annoyance when you run discovery without any arguments the usage that get's printed does not mention chainId
> 
> ```
> $ discovery discover
> Error: Not enough arguments
> 
> Usage:
>     yarn start ...................................... run the server application
>     yarn discover [project] ................. run discovery on a specific system
>     yarn discover [project] --dry-run ..... check simulated discovery bot output
>     yarn discover [project] --dev ..... run on the same block number as in .json
>     yarn invert [project] ..................... print addresses and their functions
>     yarn invert [project] --mermaid .................... print mermaid graph markup
>     yarn <start|discover> --help .......................... display this message
> ```

image

**Pull request merged**